### PR TITLE
Templates for mail notifications

### PR DESCRIPTION
**Pull request by Abyss777**
I used Velocity Engine.

Initiated engine in `Context` because we have to disable default velocity logger. Also set templates path.
It could be done in `formatMessage` function, but better do it once. I think `Context` is right place.

I think it is not good idea to have separate templates for body and subject. I made small trick.
Set variable `$Subject` in template (first line in every .vm file) and then retrieve it after merging.

[VTL](https://velocity.apache.org/engine/1.7/vtl-reference.html) is not so powerful as [JEXL](http://commons.apache.org/proper/commons-jexl/reference/syntax.html) but I think it will be able do all necessary things.

Templates leaved as was except links.

### CODE REVIEW

File: src/org/traccar/notification/NotificationFormatter.java [hunk 1]
```diff
@@ -15,223 +15,65 @@
...
-                break;
+        Template template = null;
+        try {
+            template = Context.getVelocityEngine().getTemplate(event.getType() + ".vm");
+        } catch (ResourceNotFoundException error) {
+            Log.warning(error);
         }
-        return result;
+        if (template == null) {
+            template = Context.getVelocityEngine().getTemplate("unknown.vm");
```
Comment (tananaev) [inline src/org/traccar/notification/NotificationFormatter.java]: I think we can just move this to catch. Or is there a chance that `getTemplate` returns null?

File: src/org/traccar/notification/NotificationFormatter.java [hunk 2]
```diff
@@ -15,223 +15,65 @@
...
+import org.traccar.helper.Log;
 import org.traccar.model.Device;
 import org.traccar.model.Event;
 import org.traccar.model.Position;
+import org.traccar.reports.ReportUtils;
 
 public final class NotificationFormatter {
 
-    private NotificationFormatter() {
+    public static class MailMessage {
```
Comment (tananaev) [inline src/org/traccar/notification/NotificationFormatter.java]: I think it's better to extract this class into a separate file.

File: src/org/traccar/Context.java [hunk 3]
```diff
@@ -242,6 +251,13 @@ public static void init(String[] arguments) throws Exception {
 
         if (config.getBoolean("event.enable")) {
             notificationManager = new NotificationManager(dataManager);
+            Properties velocityProperties = new Properties();
+            velocityProperties.setProperty("file.resource.loader.path",
+                    Context.getConfig().getString("mail.templatesPath", "templates/mail") + "/");
+            velocityProperties.setProperty("runtime.log.logsystem.class",
+                    "org.apache.velocity.runtime.log.NullLogChute");
```
Comment (tananaev) [inline src/org/traccar/Context.java]: What does this line mean?

Comment (tananaev) [inline src/org/traccar/Context.java]: I agree that we don't want more files.

**Comment by Abyss777** [comment]
Extracted class and moved getting "unknown.vm" to `catch`

**Comment by tananaev** [comment]
Merged, thanks for the pull request.

**Pull request merged**